python/packages/sdk/sls_sdk/lib/instrumentation/flask.py

```diff
@@ -0,0 +1,155 @@
+import re
+from sls_sdk import serverlessSdk
+from ..error import report as report_error
+from functools import partial
+from .import_hook import ImportHook
+
+_instrumenter = None
+_import_hook = ImportHook("flask")
+
+
+class Instrumenter:
+    def __init__(self, flask):
+        self._flask = flask
+        # Signals https://flask.palletsprojects.com/en/2.2.x/signals/ cover most of the
+        # use-cases, except for the case of user exceptions such as http 404.
+        # In that case, we need to override the handle_user_exception
+        # to create a span for the error. See _handle_user_exception for more details.
+        self._signals = [
+            (self._flask.appcontext_pushed, self._handle_appcontext_pushed),
+            (self._flask.request_started, self._handle_request_started),
+            (self._flask.request_finished, self._handle_request_finished),
+            (self._flask.got_request_exception, self._handle_got_request_exception),
+            (self._flask.appcontext_popped, self._handle_appcontext_popped),
+        ]
+        self._root_span = None
+        self._route_span = None
+        self._error_span = None
+        self._reported_exception = None
+
+    def sanitize_span_name(self, name):
+        if name is None:
+            return "unknown"
```

Comment:
If a span name is `unknown` do we still send it with the final trace payload?

Reply:
Yes, we still send it. I think it's the same behaviour as in nodejs SDK https://github.com/serverless/console/blob/910cbf53b638b1fa51bcdf2da54cb8666a68c27f/node/packages/sdk/lib/instrumentation/express/instrument-layer-prototype.js#L48 I couldn't find a filter that filters out "unknown" spans.